ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -31,6 +54,268 @@ let tdb_tool = !Xapi_globs.tdb_tool
 
 let debug_level = !Xapi_globs.winbind_debug_level |> string_of_int
 
+let ntlm_auth uname passwd : (unit, exn) result =
+  try
+    let args = ["--username"; uname] in
+    let _stdout =
+      Helpers.call_script ~log_output:Never ~stdin:passwd
+        !Xapi_globs.ntlm_auth_cmd args
+    in
+    Ok ()
+  with e ->
+    D.info "ntlm auth failed with uname='%s'" uname ;
+    Error e
```

Comment:
when the auth failed, the higher layer expected a `Auth_signature.Auth_failure`
 https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_session.ml#L824
cli will match the error code, and provide  messages back to user ( `Authentication failed` in this case), otherwise, just return Server Internal error,  like `The server failed to handle your request, due to an internal error. The given message may give details useful for debugging the problem.                      
message: session.login_with_password
<extra>: INTERNAL_ERROR: [ Subprocess exite`

Hope the exceptions to be raised can be checked for all the reminding APIs,  in higher layer codes or refer to PBIS,

Reply:
I added a function to extract error messages from the wbinfo stderr, and turn them into the expected exception (I believe these match up roughly with what we have for pbis)